devicetypes/fibargroup/fibaro-walli-roller-shutter-venetian.src/fibaro-walli-roller-shutter-venetian.groovy

```diff
@@ -0,0 +1,636 @@
+			return integerValue
+	}
+}
+
+private getCommandValue(preference) {
+	def parameterKey = preference.key
+	switch (preference.type) {
+		case "boolean":
+			return settings."$parameterKey" ? preference.optionActive : preference.optionInactive
+		case "boolRange":
+			def parameterKeyBoolean = parameterKey + "Boolean"
+			return !notNullCheck(settings."$parameterKeyBoolean") || settings."$parameterKeyBoolean" ? settings."$parameterKey" : preference.disableValue
+		case "range":
+			return settings."$parameterKey"
+		default:
+			return Integer.parseInt(settings."$parameterKey")
+	}
+}
+
+private isPreferenceChanged(preference) {
+	if (notNullCheck(settings."$preference.key")) {
+		def value = state.currentPreferencesState."$preference.key"
+		switch (preference.type) {
+			case "boolRange":
+				def boolName = preference.key + "Boolean"
+				if (state.currentPreferencesState."$preference.key".status == "disabled") {
+					return settings."$boolName"
+				} else {
+					return state.currentPreferencesState."$preference.key".value != settings."$preference.key" || !settings."$boolName"
+				}
+			default:
+				return state.currentPreferencesState."$preference.key".value != settings."$preference.key"
+		}
+	} else {
+		return false
+	}
+}
+
+private notNullCheck(value) {
+	return value != null
+}
+
+private statusOverrideIfNeeded(preferenceKey) {
+	switch (preferenceKey) {
+		case "forceCalibration":
+			if (state.calibrationStatus == "done") {
+				state.currentPreferencesState."$preferenceKey".status = "synced"
+			} 
+		break
+	}
+}
+
+def handleConfigurationChange(confgurationReport) {
+	switch (confgurationReport.parameterNumber) {
+		case 150: // Calibrating
+			switch(confgurationReport.scaledConfigurationValue) {
+				case 0: // "Device is not calibrated"
+					state.calibrationStatus = "notStarted"
+					break
+				case 1: // "Device is calibrated"
+                    state.calibrationStatus = "done"
+                    state.currentPreferencesState.forceCalibration.status = "synced"
+                    break
+                case 2: // "Force Calibration"
+                    state.calibrationStatus = state.calibrationStatus == "notStarted" ? "pending" : state.calibrationStatus
+                    break
+			}
+			log.info "Calibration ${state.calibrationStatus}"
+			break
+		case 151: //Operating mode
+			switch(confgurationReport.scaledConfigurationValue) {
+				case 1: // "Roller blind (with positioning)"
+					log.info "Changing device type to Fibaro Walli Roller Shutter"
+					setDeviceType("Fibaro Walli Roller Shutter")
+					break
+				case 2: // "Venetian blind (with positioning)"
+					log.info "Device is already configured as Fibaro Roller Shutter Venetian"
+					break
+				case 5: // "Roller blind with built-in driver"
+				case 6: // "Roller blind with built-in driver (impulse)"
+					log.info "Changing device type to Fibaro Walli Roller Shutter Driver"
+					setDeviceType("Fibaro Walli Roller Shutter Driver")
+					break
```

Comment:
wouldn't you need to delete the child device in these cases?

Reply:
You're right, but I've never saw it been done in any DTH, so I don't know how (even thought that such possibility wasn't provided)

Reply:
it exists: `deleteChildDevice(String deviceNetworkId, Boolean force = false) `